### PR TITLE
 diagnostics: add semgrep

### PR DESCRIPTION
**Pull request by fcsonline**
Adding support for SemGrep diagnostics.

https://semgrep.dev/

### CODE REVIEW

File: lua/null-ls/builtins/diagnostics/semgrep.lua [hunk 1]
```diff
@@ -0,0 +1,54 @@
+local h = require("null-ls.helpers")
+local methods = require("null-ls.helpers")
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/semgrep.lua:2]:
```suggestion
local methods = require("null-ls.methods")
```
```suggestion
local methods = require("null-ls.helpers")
```

File: lua/null-ls/builtins/diagnostics/semgrep.lua [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+return h.make_builtin({
+    name = "semgrep",
+    meta = {
+        url = "https://semgrep.dev/",
+        description = "Semgrep is a fast, open-source, static analysis tool for finding bugs and enforcing code standards at editor, commit, and CI time.",
+    },
+    method = DIAGNOSTICS,
+    filetypes = { "typescript", "typescriptreact", "ruby", "python", "java", "go" },
+    generator = null_ls.generator({
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/semgrep.lua:44]:
```suggestion
    generator = helpers.generator_factory({
```

Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/semgrep.lua:44]:
```suggestion
    generator = helpers.generator_factory({
```

File: doc/BUILTINS.md [hunk 3]
```diff
@@ -1094,10 +1094,27 @@ local sources = { null_ls.builtins.diagnostics.selene }
 
 #### Defaults
 
-- Filetypes: `{ "lua" }`
+- Filetypes: `{ "typescript", "typescriptreact", "ruby", "python", "java", "go" }`
```
Comment (jose-elias-alvarez) [inline doc/BUILTINS.md:1097]: Could you remove the changes to the auto generated files?

Comment (jose-elias-alvarez) [inline doc/BUILTINS.md:1097]: Could you remove the changes to the auto generated files?

File: lua/null-ls/builtins/_meta/diagnostics.lua [hunk 4]
```diff
@@ -151,6 +151,9 @@ return {
   selene = {
     filetypes = { "lua" }
   },
+  semgrep = {
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/_meta/diagnostics.lua:154]: This one too, CI will take care of updating it.

Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/_meta/diagnostics.lua:154]: This one too, CI will take care of updating it.

**Review by jose-elias-alvarez — CHANGES_REQUESTED** [review]
Looks cool, looking forward to trying this. I pointed out a couple of syntax errors that prevented this from loading, so we'll want to test it out before merging.

**Comment by fcsonline** [comment]
I'm going to polish a bit and check the style.

**Comment by fcsonline** [comment]
I think that now it looks much better. :ok_hand:

**Comment by fcsonline** [comment]
@jose-elias-alvarez  Also, taking the opportunity to say thank you for this awesome nvim plugin. :clap:

**Comment by jose-elias-alvarez** [comment]
Thanks for putting this together!

**Comment by xuliefeng** [comment]
hi  I'm using semgrep for java validation, but it's not working as expected. Add print found parsing code results results empty

image

image

**Pull request merged**